newIDE/app/src/EventsSheet/ToolbarCommands.js

```diff
@@ -38,6 +39,10 @@ const ToolbarCommands = (props: Props) => {
     handler: props.onAddCommentEvent,
   });
 
+  useCommand('TOGGLE_EVENT', true, {
+    handler: props.onToggleDisabledEvent,
```

Comment:
Here, you give the possibility to toggle disabled any events though not every event can be disabled (comment events for example).

The method `toggleDisabled` of the component `EventsSheet` should toggle an event only if it is executable (see method `_selectionCanToggleDisabled`):


```js
  toggleDisabled = () => {
    getSelectedEvents(this.state.selection).forEach(event =>
      if (event.isExecutable()) event.setDisabled(!event.isDisabled())
    );
    this._saveChangesToHistory(() => {
      if (this._eventsTree) this._eventsTree.forceEventsUpdate();
    });
  };
```

But doing that, we can see that we could add an entry to the history even though no event were changed. So we could add:

```js
  toggleDisabled = () => {
    let shouldBeSaved = false;
    getSelectedEvents(this.state.selection).forEach(event =>
      if (event.isExecutable()) {
        event.setDisabled(!event.isDisabled())
        shouldBeSaved = true;
      }
    );
    if (shouldBeSaved) {
      this._saveChangesToHistory(() => {
        if (this._eventsTree) this._eventsTree.forceEventsUpdate();
      });
    }
  };
```